specification/azsadmin/resource-manager/infrastructureinsights/Microsoft.InfrastructureInsights.Admin/preview/2016-05-01/InfrastructureInsights.json

```diff
@@ -0,0 +1,336 @@
+{
+    "swagger": "2.0",
+    "info": {
+        "version": "2016-05-01",
+        "title": "InfrastructureInsightsManagementClient",
+        "description": "The Admin Infrastructure Insights Management Client."
+    },
+    "host": "adminmanagement.local.azurestack.external",
+    "schemes": [
+        "https"
+    ],
+    "paths": {
+        "/providers/Microsoft.InfrastructureInsights.Admin/operations": {
+            "get": {
+                "x-ms-examples": {
+                    "Returns a list of support REST operations.": {
+                        "$ref": "./examples/Operations/List.json"
+                    }
+                },
+                "description": "Returns a list of support REST operations.",
+                "tags": [
+                    "InfrastructureInsights"
+                ],
+                "operationId": "Operations_List",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/OperationList"
+                        }
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": "nextLink"
+                },
+                "produces": [
+                    "application/json"
+                ],
+                "consumes": [
+                    "application/json"
+                ]
+            }
+        }
+    },
+    "definitions": {
+        "Dictionary": {
+            "description": "Dictionary of string key-value pairs.",
+            "type": "object",
+            "additionalProperties": {
+                "type": "string"
+            }
+        },
+        "BaseHealth": {
+            "description": "Models the base properties for health resource.",
+            "properties": {
+                "alertSummary": {
+                    "description": "Alert summary.",
+                    "$ref": "#/definitions/AlertSummary"
+                },
+                "healthState": {
+                    "description": "Health state.",
+                    "type": "string"
+                },
+                "namespace": {
+                    "description": "Namespace of the health resource.",
+                    "type": "string"
+                },
+                "registrationId": {
+                    "description": "Registration ID.",
+                    "type": "string"
+                },
+                "routePrefix": {
+                    "description": "Route prefix.",
+                    "type": "string"
+                }
+            }
+        },
+        "Resource": {
```

Comment:
>Resource [](start = 9, length = 8)

This definition should follow this patterns instead
``` json
    "Resource": {
      "description": "The core properties of ARM resources",
      "properties": {
        "id": {
          "readOnly": true,
          "type": "string",
          "description": "Fully qualified resource Id for the resource"
        },
        "name": {
          "readOnly": true,
          "type": "string",
          "description": "The name of the resource"
        },
        "type": {
          "readOnly": true,
          "type": "string",
          "description": "The type of the resource."
        }
      },
      "x-ms-azure-resource": true
    },
    "TrackedResource": {
      "description": "The resource model definition for a ARM tracked top level resource",
      "properties": {
        "tags": {
          "type": "object",
          "additionalProperties": {
            "type": "string"
          },
          "x-ms-mutability": [
            "read",
            "create",
            "update"
          ],
          "description": "Resource tags."
        },
        "location": {
          "type": "string",
          "x-ms-mutability": [
            "read",
            "create"
          ],
          "description": "The Azure Region where the resource lives"
        }
      },
      "allOf": [
        {
          "$ref": "#/definitions/Resource"
        }
      ]
    },
    "ProxyResource": {
      "description": "The resource model definition for a ARM proxy resource. It will have everything other than required location and tags",
      "allOf": [
        {
          "$ref": "#/definitions/Resource"
        }
      ]
    },
```

Reply:
Could you explain why?  Who would reference TrackedResource and who would reference ProxyResource?

Reply:
Tracked resource is for those resources that have location and tags (for example, a VM).
Proxy resource is for those other resources that happen to be derived of other resources, and therefore, they don't have a location/tags (for example, a VM extension).

Reply:
@mcardosos I still don't understand why you want me to add this.  Looking at fiddler all my resources have a location and tags on the returned objects.  Are you saying I should make it so nested resources cannot access those fields anymore?

Reply:
If they have location and tags, they should have an allof of trackedresource

Reply:
Done.